_pytest/warnings.py

```diff
@@ -62,7 +64,7 @@ def catch_warnings_for_item(item):
 
         for warning in log:
             msg = warnings.formatwarning(
-                warning.message, warning.category,
+                compat.safe_str(warning.message), warning.category,
```

Comment:
its required to warn the user here that his warning message is broken as it is not a str instance (after all this is a stdlib problem wrokaround)